autoscale/src/main/java/com/sequenceiq/periscope/service/AuditService.java

```diff
@@ -0,0 +1,96 @@
+package com.sequenceiq.periscope.service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.audit.AuditClient;
+import com.sequenceiq.cloudbreak.audit.model.ActorCrn;
+import com.sequenceiq.cloudbreak.audit.model.ActorService;
+import com.sequenceiq.cloudbreak.audit.model.ApiRequestData;
+import com.sequenceiq.cloudbreak.audit.model.AuditEvent;
+import com.sequenceiq.cloudbreak.audit.model.AuditEventName;
+import com.sequenceiq.cloudbreak.audit.model.ServiceEventData;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.periscope.api.model.ScalingStatus;
+import com.sequenceiq.periscope.model.json.Json;
+
+@Service
+public class AuditService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuditService.class);
+
+    private static final String CLUSTER_CRN = "clusterCrn";
+
+    private static final String AUTOSCALE_ACTION = "autoscaleAction";
+
+    private static final String TIMESTAMP = "timestamp";
+
+    private static final String AUTOSCALE_TRIGGER_STATUS = "autoscaleTriggerStatus";
+
+    @Inject
+    private AuditClient auditClient;
+
+    @Value("${info.app.version:}")
+    private String periscopeVersion;
+
+    @Async
```

Comment:
Is this configured properly with SpringBoot - i.e. thread counts, names etc (Will it spawn an unlimited number of threads, or controlled - again in case the destination service is down.)

Reply:
Yes it is configured in spring boot with default core thread size as 50 and max as 500.  In Periscope Onprem they have changed this to core 500 and max as 1000. I will submit a different PR for this change.